src/library/dlgtagfetcher.cpp

```diff
@@ -234,7 +269,21 @@ void DlgTagFetcher::reject() {
 
 void DlgTagFetcher::fetchTagProgress(const QString& text) {
     QString status = tr("Status: %1");
-    loadingStatus->setText(status.arg(text));
+    loadingProgressBar->setFormat(status.arg(text));
+    m_progressBarValue += 15;
+    loadingProgressBar->setValue(m_progressBarValue);
+}
+
+void DlgTagFetcher::progressBarSetCurrentStep() {
+    QString status = tr("Fetching track data from the MusicBrainz database");
+    loadingProgressBar->setFormat(status);
+    m_progressBarValue += m_incrementBarValueBy;
+    loadingProgressBar->setValue(m_progressBarValue);
+}
+
+// TODO(fatihemreyildiz): display the task results one by one.
+void DlgTagFetcher::progressBarSetTotalSteps(int totalRecordingsFound) {
+    m_incrementBarValueBy = 60 / totalRecordingsFound;
```

Comment:
What is 60?

Reply:
40% is the first 3 steps that is "Finger printing" - "Identifying from AcoustID" - "Retrieving metadata from MusicBrainz". 60 percent belongs to the actual fetching from Musicbrainz. 

According to Recording IDs returned from AcoustID, the progress bar can jump from 40% to 100% if just a recording ID found. Or either jump in slow steps if there are many recording IDs found.

I made 50% in the beginning but this time the steps were really small for a popular tracks. 

This percent can be changed according with your feedback.